pkg/parser/ast.go

```diff
@@ -227,6 +250,14 @@ func (b *BlockStatement) String() string {
 func (b *BlockStatement) Type() *Type {
 	return NONE_TYPE
 }
+func (b *BlockStatement) AlwaysReturns() bool {
+	return b.alwaysReturns
+}
+
+func alwaysReturns(n Node) bool {
+	r, ok := n.(Returner)
```

Comment:
:pineapple: If you wanted, you could get rid of the `Returner` interface and change this line to:

    r, ok := n.(interface{ AlwaysReturns() bool})

Essentially using an anonymous interface to check if the given type has this method. I would not normally do this, but since it is only used inside this function, you could get away with it. At the very least, I don't think `Returner` needs to be exported. It is only used internally for parsing correctness.